tests/test_estimators.py

```diff
@@ -16,12 +16,12 @@
     MSNRegressor,
     RawKNNRegressor,
 )
-from sknnr._base import IDNeighborsRegressor
+from sknnr._base import KNeighborsDFIndexCrosswalkMixin
 
 
-def get_kneighbor_estimator_classes() -> List[Type[IDNeighborsRegressor]]:
+def get_kneighbor_estimator_classes() -> List[Type[KNeighborsDFIndexCrosswalkMixin]]:
     """
-    Return classes of all supported IDNeighborsRegressor estimators.
+    Return classes of all supported KNeighborsDFIndexCrosswalkMixin estimators.
```

Comment:
No, I think you're right that this isn't a very good way to type this! I find-replaced these and didn't give them any thought.

I'm tempted to use a `Union` of all the superclasses, but I don't think that's *quite* right since that indicates it could be just one of the classes... I did a little searching, but I don't think there's any built-in type that indicates a combination of classes. I guess another option would be to create a dummy class that inherits from all three superclasses and use that type, but that seems a little like overkill. 

Any ideas or preferences?

Reply:
Well, I think `RawKNNRegressor` only inherits from `KNeighborsDFIndexCrosswalkMixin` and `KNeighborsRegressor`, so that might cause issues.  I agree that the dummy class is overkill, but can you do a dummy type that includes both `KNeighborsDFIndexCrosswalkMixin` and `KNeighborsRegressor`, e.g.:
```
DerivedEstimatorType = Type[KNeighborsDFIndexCrosswalkMixin & KNeighborsRegressor]
def get_kneighbor_estimator_classes() -> List[DerivedEstimatorType]:
    ...
```
It passes mypy on pre-commit, but I'm not sure if that will work.  (I'm a bit out of my league on this one!)

Reply:
> Well, I think RawKNNRegressor only inherits from KNeighborsDFIndexCrosswalkMixin and KNeighborsRegressor, so that might cause issues.

Good point! 

I like the dummy type solution a lot more than what I was thinking, but since we'd need a separate derived type with and without the `TransformedKNeighborsMixin`, I'm now starting to think that maybe we should just update the docstring with something more descriptive like "KNeighbors-derived estimators" and remove type hints from this function.

My feeling is that the main reason to type-hint unit tests is improving readability, since there's no benefit for users and any bugs that would be caught by mypy will of course be caught by the unit test itself. In fact, I just remembered that mypy currently [doesn't run on the test directory](https://github.com/lemma-osu/scikit-learn-knn-regression/blob/5c083e9fca9e980d9d64c185f76978f84a530ef1/.pre-commit-config.yaml#L6) which is probably why it didn't complain about my original solution! In terms of readability, this function is so simple that trying to put together accurate type hints for it might be counter-productive.

What do you think? Does that seem reasonable?

Reply:
Totally reasonable to me!

Reply:
I just realized this doesn't even need to be a function anymore! That was done with the idea that it would return a new set of instances each time it was called, but because that didn't work with `pytest.mark.parameterize` and we switched to classes, this can just be a list of classes.